File: 35. Search Insert Position.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        for (int i = 0; i<nums.size(); i++)
        {
            if (nums[i] == target)
                return i;
            else
            {
                if (target < nums[0])
                    return 0;
                else if (target > nums[nums.size() - 1])
                    return nums.size();
                else if (i > 0 && nums[i-1] < target && nums[i] > target)
                    return i;
            }
        }
        return 0;
    }
};

File: 35. Search Insert Position.py
class Solution(object):
    def searchInsert(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        for i in xrange(len(nums)):
            if nums[i] == target:
                return i
            else:
                if target < nums[0]:
                    return 0
                elif target > nums[len(nums) - 1]:
                    return len(nums)
                else:
                    if i > 0 and nums[i-1] < target and nums[i] > target:
                        return i
